app/javascript/react/components/NewShopForm.js

```diff
@@ -0,0 +1,167 @@
+    requiredFields.forEach(field => {
+      if (shopRecord[field].trim() === "") {
+        submitErrors = {
+          ...submitErrors,
+          [field]: "is blank"
+        }
+      }
+    })
+
+    setErrors(submitErrors)
+    return _.isEmpty(submitErrors)
+  }
+
+  const handleInputChange = (event) => {
+    setShopRecord({
+      ...shopRecord,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const postNewShop = async (event) => {
+    event.preventDefault()
+
+    if (validForSubmission()) {
+      try {
+        const response = await fetch('/api/v1/shops', {
+          method: "POST",
+          credentials: "same-origin",
+          headers: {
+            "Content-Type": "application/json"
+          },
+          body: JSON.stringify({ shop: shopRecord })
+        })
+        if (!response.ok) {
+          const errorMessage = `${response.status} (${response.statusText})`
+          throw new Error(errorMessage)
+        }
+        console.log("Shop was added successfully!")
+        setShouldRedirect(true)
+      } catch(err) {
+        console.error(`Error in fetch: ${err.message}`)
+      }
+    }
+  }
+
+  if (shouldRedirect) {
+    return <Redirect to='/shops' />
+  }
+
+  return(
+    <form onSubmit={postNewShop}>
+      <h2>Submit a new Bubble Tea Shop</h2>
+      <ErrorList errors={errors} />
+      <label>
+        Shop Name
+        <input 
+          type="text"
+          name="name"
+          value={shopRecord.name}
+          onChange={handleInputChange}
+        />
+      </label>
+
+      <label>
+        Address
+        <input 
+          type="text"
+          name="address"
+          value={shopRecord.address}
+          onChange={handleInputChange}
+        />
+      </label>
+
+      <label>
+        City
+        <input 
+          type="text"
+          name="city"
+          value={shopRecord.city}
+          onChange={handleInputChange}
+        />
+      </label>
+
+      <label>
+        State
+        <input 
+          type="text"
+          name="state"
+          value={shopRecord.state}
+          onChange={handleInputChange}
+        />
+      </label>
+
+      <label>
+        Zip Code
+        <input 
+          type="text"
+          name="zip"
+          value={shopRecord.zip}
+          onChange={handleInputChange}
+        />
+      </label>
+
+      <label>
+        Telephone
+        <input 
+          type="text"
+          name="telephone"
+          value={shopRecord.telephone}
+          onChange={handleInputChange}
+        />
+      </label>
+
+      <label>
+        Website
+        <input 
+          type="text"
+          name="website"
+          value={shopRecord.website}
+          onChange={handleInputChange}
+        />
+      </label>
+
+      <label>
+        Shop Picture (URL)
+        <input 
+          type="text"
+          name="picture"
+          value={shopRecord.picture}
+          onChange={handleInputChange}
```

Comment:
image uploading on the react side should be handled with a drop zone and have an S3 bucket set up - let's remove this image property for now, and in a separate feature work on image uploads